retrospective-update/monthly_products.py

```
@@ -0,0 +1,148 @@
+    cl.log('Preparing HydroSOS COGs')
+    id_pairs = pd.read_parquet(HYDROSOS_ID_PAIRS)
+    thresholds = pd.read_parquet(HYDROSOS_THRESHOLDS)
+    basins = gpd.read_parquet(HYDROSOS_BASINS)
+    hydrosos_color_map = {
+        1: '#cd233f',
+        2: '#ffa885',
+        3: '#e7e2bc',
+        4: '#8eceee',
+        5: '#2c7dcd',
+    }
+    color_to_rgb = lambda hex_color: tuple(int(hex_color[i:i + 2], 16) for i in (1, 3, 5))
```

Ruff (0.12.2)

Do not assign a `lambda` expression, use a `def`

Rewrite `color_to_rgb` as a `def`

(E731)